### ISSUE TITLE
negative losses?

### ISSUE THREAD

**Issue by matthewvowels1** [gabloa/cont_bern#1]
Great work, thank you very much for the contribution. 

I've attempted a pytorch version of the loss computation. However, I get some negative losses when either reconstruction or targets (or both) have values around 0.5. I've probably messed up somewhere, but thought I'd post this in case you have some useful experience to kindly share : ]  (I also realise such artificial data is not reflective of a continuous bernoulli dist)

`def atanh(x):
    return 0.5*torch.log((1+x)/(1-x))
def cont_bern_log_norm(lam, l_lim=0.49, u_lim=0.51):
    cut_lam = torch.where((torch.lt(lam, l_lim) | torch.gt(lam, u_lim)) ,  lam,  l_lim * torch.ones_like(lam))
    log_norm = torch.log(torch.abs(2.0 * atanh(1 - 2.0 * cut_lam))) - torch.log(torch.abs(1 - 2.0 * cut_lam))
    taylor = np.log(2.0) + 4.0 / 3.0 * (lam - 0.5)**2 + 104.0 / 45.0 * (lam-0.5)**4
    return torch.where((torch.lt(lam, l_lim) | torch.gt(lam, u_lim)), log_norm, taylor)

recon_x = torch.randn((10,3,64,64))
recon_x = torch.clamp(recon_x, 1e-4, 1-1e-4)/2
x = torch.randn((10,3,64,64))
x = torch.clamp(x, 1e-4, 1-1e-4) /2
log_norm_const = cont_bern_log_norm(recon_x)
recon = -torch.sum(x * torch.log(recon_x) + (1-x) * torch.log(1-recon_x) + log_norm_const).div(10)`
